core/tx_pool.go

```diff
@@ -673,8 +675,8 @@ func (pool *TxPool) promoteExecutables(state *state.StateDB) {
 			delete(pool.all, hash)
 			pool.priced.Removed()
 		}
-		// Drop all transactions that are too costly (low balance)
-		drops, _ := list.Filter(state.GetBalance(addr))
+		// Drop all transactions that are too costly (low balance or out of gas)
+		drops, _ := list.Filter(state.GetBalance(addr), gaslimit)
```

Comment:
I don't follow.. Normally, the `gasLimit()` will equal the transaction with most gas.. so how can that filter anything?